### ISSUE TITLE
"Synced at Invalid Date" is wrongly displayed when logging in the Sync Account

### ISSUE THREAD

**Issue by SoftVision-CarmenFat** [mozilla/notes#323]
[Affected versions]:
- Firefox Notes 2.0.0a4
- Firefox 56.0 and up

[Affected Platforms]:
- All Windows
- All Mac
- All Linux

[Prerequisites]:
- Have a Firefox profile with the latest "Firefox Notes" add-on version (2.0.0a4) installed.
- Have a valid Sync Account.

[Steps to reproduce]:
1. Open the browser with the profiles from prerequisites.
2. Click the "Sync your Notes" button from the bottom of the sidebar.
3. Enter your credentials and confirm the e-mail.
4. Observe the string from the bottom part of the Sidebar.

[Expected result]:
- "Synced at (hour)" is displayed.

[Actual result]:
- "Synced at Invalid Date" string is displayed.

[Notes]:
- The following error is displayed in the Browser console:
![browser console](https://user-images.githubusercontent.com/26000308/31616574-b8d39304-b295-11e7-9d07-cc0edaac0950.png)

- Attached a screen recording of the issue:

![invalid date](https://user-images.githubusercontent.com/26000308/31616583-bcada14a-b295-11e7-8d9d-90749a41f482.gif)

**Comment by SoftVision-EmilPasca** [comment]
Refs: [1411646](https://bugzilla.mozilla.org/show_bug.cgi?id=1411646)

**Comment by Natim** [comment]
```
TypeError: key.k is undefined
Trace de la pile :
shared_key@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/sync.js:15:21
encrypt@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/sync.js:23:58
encode@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/sync.js:66:30
async*[21]</_encodeRecord/</<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto.js:6477:24
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:596:1731
[200]</notify/<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:596:1870
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:193:343
MutationCallback*[65]</module.exports@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:193:504
[200]<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:596:450
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:617
s/<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:676
[297]<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:886:4279
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:617
s/<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:676
[1]</<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:3:113
[1]<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:2:2
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:617
s/<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:676
[312]<@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:930:997
s@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:617
e@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:796
@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:382
@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:321
@moz-extension://b8d86afd-faaf-4e92-80e5-2577241d8f7f/vendor/kinto-http.js:1:2
  sync.js:228:9
```

**Comment by Natim** [comment]
Fixed with f2fc45342248c2a6856b7dd842d8bf0a8c902118

**Comment by Natim** [comment]
It is still the case if Kinto is empty
